synapse/storage/state.py

```diff
@@ -469,20 +518,30 @@ def _get_some_state_from_cache(self, group, types):
         missing state.
 
         Args:
-            group: The state group to lookup
-            types (list): List of 2-tuples of the form (`type`, `state_key`),
-                where a `state_key` of `None` matches all state_keys for the
-                `type`.
+            group(int): The state group to lookup
+            types(list[str|None, str|None]): List of 2-tuples of the form
+                (`type`, `state_key`), where a `state_key` of `None` matches all
+                state_keys for the `type`. Presence of type of `None` indicates
+                that types not in the list should not be filtered out.
         """
         is_all, known_absent, state_dict_ids = self._state_group_cache.get(group)
 
         type_to_key = {}
         missing_types = set()
 
+        include_other_types = False
+
         for typ, state_key in types:
             key = (typ, state_key)
+
+            if typ is None:
```

Comment:
see the above comments about obscure special-cases, but if we're going to stick with this, could it be made to check for `(None, None)` rather than `(None, *)` for consistency with `_get_state_groups_from_groups`

Reply:
fixed by `filtered_types`